pip/req/req_set.py

```diff
@@ -423,6 +436,15 @@ def _check_skip_installed(self, req_to_install, finder):
                         req_to_install.conflicts_with = \
                             req_to_install.satisfied_by
                     req_to_install.satisfied_by = None
+
+            # Figure out a nice message to say why we're skipping this.
+            if best_installed:
+                skip_reason = 'already up-to-date'
+            elif self.upgrade_strategy == "non-eager":
+                skip_reason = 'not upgraded as not directly required'
```

Comment:
This gets appended to "Requirement already" instead of to "Requirement". Now it looks a bit weird: ``Requirement already not upgraded as not directly required``